src/main/scala/com/fulcrumgenomics/rnaseq/EstimateRnaSeqInsertSize.scala

```diff
@@ -90,12 +89,20 @@ class EstimateRnaSeqInsertSize
     val in                  = SamSource(input)
     val refFlatSource       = RefFlatSource(refFlat, Some(in.dict))
     val counters            = pairOrientations.map { pairOrientation => (pairOrientation, new NumericCounter[Long]()) }.toMap
-    val filter              = new AggregateFilter(EstimateRnaSeqInsertSize.filters(minimumMappingQuality=minimumMappingQuality, includeDuplicates=includeDuplicates).asJava)
     var numReadPairs        = 0L
     val recordIterator      = in.iterator.filter { rec =>
       progress.record(rec)
       if (rec.paired && rec.firstOfPair) numReadPairs += 1
-      !filter.filterOut(rec.asSam)
+
+      rec.firstOfPair && // so templates are considered only once
+        rec.pf && // don't want low quality reads
+        rec.paired && // only want paired reads
+        !rec.secondary && !rec.supplementary && // no secondary or supplementary
+        rec.mapq >= minimumMappingQuality && // decent mapping quality
+        (includeDuplicates || !rec.duplicate) && // no duplicates, unless specified
+        rec.mapped && rec.mateMapped && // both ends of a pair are mapped
+        rec.refIndex == rec.mateRefIndex && // both ends of a pair map to the same contig
+        (!skipMissingMateCigar || rec.mateCigar.isDefined) // skip records with missing mate cigars if specified
```

Comment:
moved filters out of the old "htsjdk" filter style, so this could use a double-check